specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json

```diff
@@ -877,6 +897,39 @@
         }
       }
     },
+    "ServicePrincipal": {
```

Comment:
This looks a lot like SystemAssignedServiceIdentity in
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/managedidentity.json
except this one adds clientId. Why is there a divergence?

Reply:
sorry, the owner is not located. 
can we fix this issue in the later PR?

Reply:
Getting all services using the same definitions for system assigned identity helps ensure API consistency across Azure. And comparing the types manually makes for a much harder review process and raises the risk of breaking changes because inconsistences exist. For example you don't use the 'uuid' format specifier whereas common types. (Is that going to be considered breaking? I'm not even sure.) Is that what you prefer?

If you can get the definition right this first time, then future breaking changes checks could at least be helping you ensure you are keeping it consistent, even though your auto-generated SDK isn't generating it the same yet, or even though you go back to autogenerating it without common types, you should get property level consistency checking.

Reply:
This ServicePrincipal object is introduced in scope of windows authentication feature. It is the representation of AAD application, which will be used for Kerberos authentication.

Regarding to 'uuid' types: Here we are using strings, as for managing AAD application we are using MSAL library which returns/expects strings for these parameters. We stored these parameters as string, to get rid of uuid <-> string conversions.

Reply:
@vmanukyan1 That's actually a rather good practical criticism of using UUID format specifier. I'm OK with not-fixing this for now based on that.

/cc @pilor